tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,21 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_orders('CFAK')
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_asarray_chkfinite(self, xp, dtype, order):
+        a = [0, 4, 0, 5]
+        return xp.asarray_chkfinite(a, dtype=dtype, order=order)
+
+    @testing.for_orders('CFAK')
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite_non_finite_vals(self, dtype, order):
+        a = [-numpy.inf, 0., numpy.inf, numpy.nan]
+        for xp in (numpy, cupy):
+            with pytest.raises(ValueError):
```

Comment:
Hey Khushi @khushi-411 
I believe you can check the error message as well. 


```
with pytest.raises(ValueError, match='Expected error message'):
```

This is a good practice and ensures exact error is raised. You can use Regex etc but passing a string works well too.

Reply:
Hi, @oke-aditya!
Oo yes! Thanks for pointing out (: I'll keep a note!

Reply:
Hmmm I missed your [previous observation](https://github.com/cupy/cupy/pull/6275#discussion_r775804617) and so the CI failed, let's fix it...
```suggestion
            if xp.issubdtype(dtype, xp.integer):
                error = OverflowError
            else:
                error = ValueError
            with pytest.raises(error):
```

Reply:
Here, we set `OverflowError` for all the dtypes lower/equal than integer type and the rest raises the `ValueError`.
I want to know more about the fixes you made and the error raised. The error message in the CI is:
```
=================================== FAILURES =================================== |  
-- | --
00:48:25.963796 STDOUT 2232] | _______________ TestKind.test_asarray_chkfinite_non_finite_vals ________________ |  
00:48:25.963800 STDOUT 2232] |   |  
00:48:25.963892 STDOUT 2232] | self = <cupy_tests.manipulation_tests.test_kind.TestKind testMethod=test_asarray_chkfinite_non_finite_vals> |  
00:48:25.963896 STDOUT 2232] | dtype = <class 'numpy.int8'>, order = 'C' |  
00:48:25.963902 STDOUT 2232] |   |  
00:48:25.963940 STDOUT 2232] | @testing.for_orders('CFAK') |  
00:48:25.963943 STDOUT 2232] | @testing.for_all_dtypes() |  
00:48:25.963950 STDOUT 2232] | def test_asarray_chkfinite_non_finite_vals(self, dtype, order): |  
00:48:25.964010 STDOUT 2232] | a = [-numpy.inf, 0., numpy.inf, numpy.nan] |  
00:48:25.964015 STDOUT 2232] | for xp in (numpy, cupy): |  
00:48:25.964020 STDOUT 2232] | with pytest.raises(ValueError): |  
00:48:25.964056 STDOUT 2232] | >               xp.asarray_chkfinite(a, dtype=dtype, order=order) |  
00:48:25.964058 STDOUT 2232] |   |  
00:48:25.964065 STDOUT 2232] | a          = [-inf, 0.0, inf, nan] |  
00:48:25.964113 STDOUT 2232] | dtype      = <class 'numpy.int8'> |  
00:48:25.964118 STDOUT 2232] | order      = 'C' |  
00:48:25.964126 STDOUT 2232] | self       = <cupy_tests.manipulation_tests.test_kind.TestKind testMethod=test_asarray_chkfinite_non_finite_vals> |  
00:48:25.964190 STDOUT 2232] | xp         = <module 'numpy' from '/opt/pyenv/versions/3.9.6/lib/python3.9/site-packages/numpy/__init__.py'>
```
Thank you!!

cc: @leofang

Reply:
In part of this test we convert infs/nans to ints. But they are not compatible and an `OverflowError` would be raised if `dtype` is of any integer type, so we need to branch out to let pytest catch different errors.

Looks like when `dtype` is bool nothing would be raised, so this test still failed. Let me suggest another approach shortly. (You can see why I said we need to check those corner cases 🙂)

Reply:
Oo, that's the reason, Got It! Thanks!
> You can see why I said we need to check those corner cases 🙂

Yes, it's very much interesting 🙂